src/libraries/Native/Windows/CMakeLists.txt

```diff
@@ -11,15 +11,19 @@ SET (CMAKE_ASM_MASM_FLAGS                   "${CMAKE_ASM_MASM_FLAGS} /ZH:SHA_256
 SET (CMAKE_INSTALL_PREFIX                   $ENV{__CMakeBinDir})
 SET (CMAKE_INCLUDE_CURRENT_DIR              ON)
 SET (CMAKE_SHARED_LIBRARY_PREFIX            "")
-SET (__LinkArgs $ENV{__LinkArgs})
+
+set(__SharedLinkArgs)
+set(__LinkArgs)
+if ($ENV{__LinkArgs})
+    SET (__LinkArgs $ENV{__LinkArgs})
```

Comment:
Would unconditional ```SET (__LinkArgs $ENV{__LinkArgs})``` have the same effect?

Reply:
No, it would not. When the env is not defined (which is the normal state), this sets the variable to an empty string. When we add further options to the list, it remains there as empty string list entry and that breaks build when that is passed to the linker.

Reply:
Makes sense.